spec/services/fees/create_instant_service_spec.rb

```diff
@@ -0,0 +1,229 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe Fees::CreateInstantService, type: :service do
+  subject(:fee_service) { described_class.new(charge:, event:) }
+
+  let(:organization) { create(:organization) }
+  let(:billable_metric) { create(:billable_metric, organization:) }
+  let(:customer) { create(:customer, organization:) }
+  let(:plan) { create(:plan, organization:) }
+  let(:subscription) { create(:active_subscription, organization:, customer:, plan:) }
+
+  let(:group) { nil }
+
+  let(:charge) { create(:standard_charge, :instant, billable_metric:, plan:) }
+  let(:event) { create(:event, subscription:, customer:) }
+
+  describe '#call' do
+    let(:aggregation_result) do
+      BaseService::Result.new.tap do |result|
+        result.aggregation = 9
+        result.count = 4
+        result.options = {}
+      end
+    end
+
+    let(:charge_result) do
+      BaseService::Result.new.tap do |result|
+        result.amount = 10
+        result.count = 1
+        result.units = 9
+      end
+    end
+
+    before do
+      allow(BillableMetrics::InstantAggregationService).to receive(:call)
+        .with(billable_metric:, boundaries: Hash, group:, properties: Hash, event:)
+        .and_return(aggregation_result)
+
+      allow(Charges::ApplyInstantChargeModelService).to receive(:call)
+        .with(charge:, aggregation_result:, properties: Hash)
+        .and_return(charge_result)
+    end
+
+    it 'creates a fee' do
+      result = fee_service.call
+
+      aggregate_failures do
+        expect(result).to be_success
+
+        expect(result.fees.count).to eq(1)
+        fee = result.fees.first
+        expect(fee.subscription).to eq(subscription)
+        expect(fee.charge).to eq(charge)
+        expect(fee.amount_cents).to eq(10)
+        expect(fee.amount_currency).to eq('EUR')
+        expect(fee.vat_rate).to eq(20.0)
+        expect(fee.vat_amount_cents).to eq(2)
+        expect(fee.vat_amount_currency).to eq('EUR')
+        expect(fee.fee_type).to eq('instant_charge')
+        expect(fee.invoiceable).to eq(charge)
+        expect(fee.units).to eq(9)
+        expect(fee.properties).to be_present
+        expect(fee.events_count).to eq(1)
+        expect(fee.group_id).to be_nil
```

Comment:
what about having one expectation by using `expect(fee).to have_attributes(...)`?